pkg/srv/mstream/mstream.go

```diff
@@ -170,6 +176,9 @@ func (s *MStreamServer) Run() error {
 			}
 		}(s.writerStateChs[deviceName], s.writerChs[deviceName])
 
+		//Run parse event from event builders for api
+		go ParseEventsForApi(s.apiProgressByteCh, s.apiProgressJsonCh, s.apiProgressReqCh)
```

Comment:
You run multiple parsers (one per device), but all event builders put event data in the same channel.

Reply:
I am fixed it